File: python/potega.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#  potega.py
# 2^3 = 2 * 2 * 2
# 4^4 = 4 * 4 * 4 *4 

def czy_naturalna(a):
    if a.isdigit():
        return True
    return False

def potega_it(a, n):
    wynik = 1
    for i in range(n):
        wynik = wynik * a
    return wynik

def potega_rek(a, n):
    # a^n = a^(n-1) * a
    if n == 0 :
        return 1
    return potega_rek(a, n-1) * a

def main(args):
    assert potega_rek(2, 0) == 1
    assert potega_rek(2, 1) == 2
    assert potega_rek(0, 10) == 0
    assert potega_rek(1, 10) == 1
    assert potega_rek(3, 3) == 27
    
    a = input("Podaj podstawę: ")
    n = input("Podaj wykładnik: ")
    if not czy_naturalna(a) or not czy_naturalna(n):
        print("Błędne dane!")
        return 0
    print("{} do potęgi {} = {}".format(a, n, potega_it(int(a), int(n))))
    return 0

if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))

File: cpp/potega.cpp
#include <iostream>
using namespace std;

// x0 = 1 dla x=/0
// x1 = x
// xn = x * ... * x(n-czynników)

float potega_it(float x, int n) {
	float potega = 1;
	for (int i=0; i < n; i++){
		potega = potega * x;
	}
	return potega;
}


// x0 = 1 dla x=/0
// xn = x(n-1) * x
float potega_re(float x, int n){
	if (n == 0)
		return 1;
	else
	return potega_re(x, n-1) * x;
}

int main(int argc, char **argv)
{
	float x; // podstawa
    int n; // wykładnik
	cout << "Podaj podstawę: " << endl;
	cin >> x;
    cout << "Podaj wykładnik: " << endl;
	cin >> n;
	cout << "Potęga: " << potega_it(x, n) << endl;
    cout << "Potęga: " << potega_re(x, n) << endl;
	return 0;
}
